app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -942,22 +1100,26 @@ public void onPageFinished(boolean isSecure) {
                 siteIdentity.setImageLevel(SITE_LOCK);
             }
             String urlToBeInserted = getUrl();
-            if (!getUrl().equals(this.failingUrl) && !urlToBeInserted.equals(lastInsertedUrl) && getWebView() != null) {
-                getWebView().insertBrowsingHistory();
+            if (!getUrl().equals(this.failingUrl) && !urlToBeInserted.equals(lastInsertedUrl)) {
+                tabsSession.getCurrentTab().getTabView().insertBrowsingHistory();
                 lastInsertedUrl = urlToBeInserted;
             }
         }
 
         @Override
-        public void onURLChanged(final String url) {
+        public void onTabCountChanged(int count) {
+            tabCounter.setCountWithAnimation(count);
+        }
+
+        @Override
+        public void onURLChanged(@NonNull Tab tab, final String url) {
             updateURL(url);
         }
 
         @Override
-        public void onProgress(int progress) {
```

Comment:
progressView.setProgress is setting progress even if current focus is not this tab.